internal/driver/rest/api.go

```diff
@@ -55,35 +55,39 @@ func (a *API) HandlePostRegister(w http.ResponseWriter, r *http.Request) {
 }
 
 func (a *API) HandlePostLogin(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodPost {
```

Comment:
Why did you need to check POST method? 

This handle should handle the intended request without dealing with any HTTP method checking. It keeps the function clean.